pkg/api/conversion/conversion.go

```diff
@@ -0,0 +1,55 @@
+package conversion
+
+import (
+	"encoding/json"
+)
+
+const (
+	Prefix = "conversion.internal.dynatrace.com/"
+
+	AutoUpdateKey = Prefix + "auto-update"
+)
+
+type Field[T any] struct {
+	data map[string]string
+	name string
+}
+
+func (f Field[T]) Key() string {
```

Comment:
this key method is unused outside of this package, which means it can be removed.

an alternative would be to remove the AutoUpdateKey constant (or make it private) in favor of calls like this `dk.RemovedFields().AutoUpdate.Key()`. this would ensure that these keys are always accessed through the removed fields at the cost of verbosity.